### ISSUE TITLE
Empty Css File for Some Sub domain

### ISSUE THREAD

**Issue by Boxjan** [nextcloud/server#20746]
---
name: 🐛 Bug report
about: Help us improving by reporting a bug
labels: bug, 0. Needs triage
---

### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
The server is set in US, and I set up cloudflare reverse proxy and set up nginx reverse proxy in some where, with different sub-domain.
But problem happened, After I upgrade to 18.0.4 from 18.0.3 ,I notice when I use reverse proxy to visit, the style file is empty, but direct connect is well. There are no log about it, then I compare the different of the style, I found that different domain will get different style file path. When I use nginx  reverse proxy visit it, the css file path will get empty style files, all of them. After I change host of header in reverse proxy config, the problem was solved. So, I think may be have something wrong in it.

I'm so sorry that english is not my fisrt language.
### Expected behaviour
Got Empty css file when visit by some sub-domain
### Actual behaviour
get no empty css file when visit by any sub-domain

### Server configuration

**Operating system:**
Ubuntu
**Web server:**
Nginx + Fpm
**Database:**
Pgsql 11
**PHP version:**
php 7.3.17
**Nextcloud version:** (see Nextcloud admin page)
18.0.4
**Updated from an older Nextcloud/ownCloud or fresh install:**
18.0.3
**Where did you install Nextcloud from:**
Docker
**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
IEnabled:
  - accessibility: 1.4.0
  - activity: 2.11.0
  - admin_audit: 1.8.0
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - contacts: 3.3.0
  - dav: 1.14.0
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_external: 1.9.0
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - firstrunwizard: 2.7.0
  - gpxpod: 4.2.1
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - mail: 1.3.4
  - maps: 0.1.6
  - news: 14.1.6
  - nextcloud_announcements: 1.7.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - password_policy: 1.8.0
  - photos: 1.0.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - recommendations: 0.6.0
  - richdocuments: 3.5.3
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - spreed: 8.0.8
  - support: 1.1.0
  - survey_client: 1.6.0
  - systemtags: 1.8.0
  - text: 2.0.0
  - twofactor_backupcodes: 1.7.0
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
Disabled:
  - bruteforcesettings
  - encryption
  - theming
  - user_ldap

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>
domain have been change.
```
{
    "system": {
        "objectstore": {
            "class": "OC\\Files\\ObjectStore\\S3",
            "arguments": {
                "bucket": "***REMOVED SENSITIVE VALUE***",
                "autocreate": true,
                "key": "***REMOVED SENSITIVE VALUE***",
                "secret": "***REMOVED SENSITIVE VALUE***",
                "hostname": "sfo2.digitaloceanspaces.com",
                "port": 443,
                "use_ssl": true,
                "region": "optional",
                "use_path_style": true
            }
        },
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": true
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "cloud.example.com",
            "1": "hk.cloud.example.com",
            "2": "cf.cloud.example.com"
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "18.0.4.2",
        "overwriteprotocol": "https",
        "overwrite.cli.url": "https:\/\/cloud.example.com",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "app_install_overwrite": [
            "bruteforcesettings"
        ]
    }
}

```
</details>

**Are you using external storage, if yes which one:** 
S3 

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:**
Chrome
**Operating system:**
Os X, Windows 10
#### Web server error log
<details>
<summary>Web server error log</summary>

```
Nothing Error
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"8rxTIA0401bmj2Aeateo","level":2,"time":"2020-04-30T11:27:22+00:00","remoteAddr":"2001:470:18:f4b::2","user":"boxjan","app":"core","method":"GET","url":"/apps/files/","message":"Failed to compile and/or save /var/www/html/apps/files/css/merged.scss","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36","version":"18.0.4.2","id":"5eaabf4583e59"}
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by csxdotcc** [comment]
hi!
as you didn't mention it, did you set up reverse proxy configuration like described in the docs?

https://docs.nextcloud.com/server/18/admin_manual/configuration_server/reverse_proxy_configuration.html


i also used this thread/posts at the time:

https://help.nextcloud.com/t/nginx-reverse-proxy-what-to-write-in-nextclouds-config-php/9149/8
https://help.nextcloud.com/t/nginx-reverse-proxy-what-to-write-in-nextclouds-config-php/9149/14

**Comment by Boxjan** [comment]
Hi,
I just solved the problem, after I **clean everything in the redis** which is only used by nextcloud, the problem is gone. 

I don't think the problem in my reverse proxy, because I found that direct connet will not show some style, but connet by proxy (Nginx/Cloudflare) is show well, it seem like you visit different sub-domain will get different theme, but only one theme in my server. After clean the redis, no problem any more.

So I still think may be there have problem about style file cache or other things.
